File: 0057-insert-interval/0057-insert-interval.java
class Solution {
    public int[][] insert(int[][] intervals, int[] newInterval) {
        List<int[]>list = new ArrayList<>();
        int start = newInterval[0];
        int end = newInterval[1];
        for(int[] interval:intervals){
            if(interval[0] > end){
                list.add(new int[]{start,end});
                start = interval[0];
                end = interval[1];
            }else{
                if(interval[1] < start){
                    list.add(interval);
                }else {
                    start = Math.min(start, interval[0]);
                     end = Math.max(end,interval[1]);
                    }
            }
        }
        list.add(new int[]{start,end});
        return list.toArray(new int[list.size()][]);
    }
}

File: 0057-insert-interval/0057-insert-interval.cpp
class Solution {
public:
    vector<vector<int>> insert(vector<vector<int>>& intervals, vector<int>& newInterval) {
        vector<vector<int>>ans;
        int start = newInterval[0],end=newInterval[1];
        for(auto it:intervals){
            if(it[0]>end){
                vector<int>temp(2);
                temp[0]=start;
                temp[1]=end;
                ans.push_back(temp);
                start=it[0];
                end=it[1];
            }else{
                if(start>it[1]){
                    ans.push_back(it);
                }else{
                    start=min(start,it[0]);
                    end=max(end,it[1]);
                }
            }
        }
        vector<int>temp(2);
        temp[0]=start;
        temp[1]=end;
        ans.push_back(temp);
        return ans;
    }
};